### PR TITLE
Replace memory allocation point of ValueTrack correctly in AnimationMixer

### PR DESCRIPTION
**Pull request by alesliehughes**

### CODE REVIEW

File: scene/animation/animation_mixer.cpp [hunk 1]
```diff
@@ -638,7 +638,13 @@ bool AnimationMixer::_update_caches() {
...
 				switch (track_type) {
 					case Animation::TYPE_VALUE: {
-						TrackCacheValue *track_value = memnew(TrackCacheValue);
+						TrackCacheValue *track_value;
+
+						// If a value track without a key is cached first, the initial value cannot be determined.
+						// It is a corner case, but which may cause problems with blending.
+						ERR_CONTINUE_MSG(anim->track_get_key_count(i) == 0, "AnimationMixer: '" + String(E) + "', Value Track:  '" + String(path) + "' must have at least one key to cache for blending.");
+
+						track_value = memnew(TrackCacheValue);
```
Comment (TokageItLab) [inline scene/animation/animation_mixer.cpp:647]:
```suggestion
						// If a value track without a key is cached first, the initial value cannot be determined.
						// It is a corner case, but which may cause problems with blending.
						ERR_CONTINUE_MSG(anim->track_get_key_count(i) == 0, "AnimationMixer: '" + String(E) + "', Value Track:  '" + String(path) + "' must have at least one key to cache for blending.");

						TrackCacheValue *track_value = memnew(TrackCacheValue);
```

**Review by TokageItLab — CHANGES_REQUESTED** [review]

File: scene/animation/animation_mixer.cpp [hunk 2]
```diff
@@ -638,6 +638,11 @@ bool AnimationMixer::_update_caches() {
 
 				switch (track_type) {
 					case Animation::TYPE_VALUE: {
+
```
Comment (TokageItLab) [inline scene/animation/animation_mixer.cpp:641]:
```suggestion
```

**Review by TokageItLab — CHANGES_REQUESTED** [review]

**Comment by alesliehughes** [comment]
I'll fix the commit up shortly. Didn't mean to create a second commit.

**Review by TokageItLab — APPROVED** [review]

**Comment by TokageItLab** [comment]
Okay, please squash later.

**Comment by akien-mga** [comment]
@alesliehughes It's not obvious to me what this does, could you add more details on what the bug is in the PR description?

**Comment by alesliehughes** [comment]
> @alesliehughes It's not obvious to me what this does, could you add more details on what the bug is in the PR description?

When a animation track doesn't have an keys, it's possible that we leak memory due the ERR_CONTINUE_MSG macro usage. 

By checking the error condition first, we avoid a allocated and thus the leak.

I'll update the patch with the above.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by YuriSizov** [comment]
Cherry-picked for 4.2.2.

**Pull request merged**